openeo_driver/errors.py

```
@@ -215,6 +215,17 @@ def __init__(self, filename: str):
         super().__init__(message=self.message.format(file=filename))
 
 
+class FileExpiredException(OpenEOApiException):
```

New errors in this file have first to be defined through https://github.com/Open-EO/openeo-api/blob/master/errors.json

see https://github.com/Open-EO/openeo-api/issues/379

---

in the mean time, you can just raise a generic exception:

```python
        raise OpenEOApiException(
            status_code=410,
            code="FileExpired",
            message="File '{file}' has expired".format(file=filename)"
         )
```

---

Interesting suggestion from https://github.com/Open-EO/openeo-api/issues/379#issuecomment-819456415:
raise it with something like:

```python
raise OpenEOApiException( 
    status_code=410, 
    code="AssetLinkExpired", 
    message="Batch job asset link has expired. Request the batch job results again for fresh asset links" ,
)
```